chalk-engine/src/logic.rs

```diff
@@ -103,6 +103,25 @@ impl<C: Context> Forest<C> {
         }
     }
 
+    pub(super) fn any_future_answer(
+        &mut self,
+        table: TableIndex,
+        mut test: impl FnMut(&mut C::CanonicalExClause) -> bool,
+    ) -> bool {
+        let mut strands = self.tables[table].strands_mut();
+
```

Comment:
and here if there is an answer with that index we would invoke `test`:

```rust
         // First, check for a tabled answer.
         if self.tables[table].answer(answer).is_some() {
             info!("answer cached = {:?}", self.tables[table].answer(answer));
             return test(...);
    }
```

Reply:
but there is a problem because it is not a canonical ex clause, but rather a substitution